apps/indexer/src/indexers/validatorIndexer.ts

```diff
@@ -169,4 +173,42 @@ export class ValidatorIndexer extends Indexer {
   afterEveryBlock(): Promise<void> {
     return Promise.resolve(undefined);
   }
+
+  /**
+   * Handles empty or invalid delegatorAddress by using validatorAddress as fallback
+   * Same logic as genesis validator creation
+   */
+  private getAccountAddress(delegatorAddress: string, validatorAddress: string, isGenesis: boolean = false): string {
+    if (!delegatorAddress || delegatorAddress.trim() === "") {
+      console.warn(`Empty delegatorAddress for validator ${validatorAddress}, using validatorAddress as accountAddress`);
+      return validatorAddress;
+    }
+
+    // For genesis validators, handle bech32 decoding
+    if (isGenesis) {
+      try {
+        // Try to decode the delegator address as bech32
+        return toBech32(activeChain.bech32Prefix, fromBech32(delegatorAddress).data);
+      } catch (error) {
+        // If bech32 decoding fails, the delegator_address might already be in the correct format
+        // or it might be a different type of address. Use it as-is.
+        console.warn(
+          `Failed to decode delegator_address "${delegatorAddress}" as bech32, using as-is:`,
+          error instanceof Error ? error.message : String(error)
+        );
+        return delegatorAddress;
+      }
+    }
+
+    return delegatorAddress;
+  }
+
+  /**
+   * Converts commission rate from Cosmos SDK format to decimal format
+   * Handles both integer format (18 decimal places) and decimal format
+   */
+  private convertCommissionRate(rate: string): number {
+    const parsedRate = parseFloat(rate);
+    return parsedRate > 1 ? parsedRate / 1e18 : parsedRate;
+  }
```

Comment:
here we will loose precision and will have values like `61.000000000000000001`. Money should be either represented as string or as a BigInt in minor units